### PR TITLE
Support removing a note from rotation

### LINKED ISSUE #9: Add ability to "suppress" note, so it doesn't appear in the normal rotation
Probably need some kind of "suppressed note" text files, with file names of the notes (one per line) that have been suppressed.

### PR DESCRIPTION
**Pull request by glenshelly**
Also add .gitignore and additional sample file

Closes #9

### CODE REVIEW

File: totes [hunk 1]
```diff
@@ -21,30 +30,42 @@ get_note_to_render() {
...
+    if ! [[ -f "$suppressed_files_registry_spec" ]]; then
+      # the suppressed_file_registry must not be empty
+      echo "dummy-text-required-here" > "$suppressed_files_registry_spec"
+    fi
+
     # ref - https://github.com/koalaman/shellcheck/issues/1247
     # shellcheck disable=2012
-    note_to_render="$(ls -At -- "$notes_path"/*.md | tail -n 1)"
+    # shellcheck disable=2010
+    note_to_render="$(ls -At -- "$notes_path"/*.md | grep -vFf "$suppressed_files_registry_spec" | tail -n 1)"
```
Comment (glenshelly) [inline totes]: fwiw, I'm working on issue #8, where I'm planning to change this line to the following:

`      ls -Art -- "$notes_path"/**/*.md | grep -m 1 -vFf "$suppressed_files_registry_spec"`

That is, reverse the ls listing, glob into subdirectories, and only return the first grep match, which obviates the need to pipe through 'tail' (while potentially being faster)

File: totes [hunk 2]
```diff
@@ -9,6 +9,15 @@ get_render_command() {
   fi
 }
 
+get_notes_path() {
+  echo "${XDG_DATA_HOME:-$HOME/.local/share}/totes"
+}
+
+get_suppressed_files_registry_spec() {
```
Comment (nathanshelly) [inline totes]: what does the `spec` mean?

File: totes [hunk 3]
```diff
@@ -29,22 +38,34 @@ get_note_to_render() {
...
     fi
 
+    suppressed_files_registry_spec="$(get_suppressed_files_registry_spec)"
+    if ! [[ -f "$suppressed_files_registry_spec" ]]; then
+      # the suppressed_file_registry must not be empty
+      echo "dummy-text-required-here" > "$suppressed_files_registry_spec"
+    fi
+
     # ref - https://github.com/koalaman/shellcheck/issues/1247
     # shellcheck disable=2012
```
Comment (nathanshelly) [inline totes]: this disable and ref no longer seems to apply based on the linter

> after reading [the lint](https://github.com/koalaman/shellcheck/wiki/SC2012) I'm not exactly sure why but  🤷‍♂

File: totes (hunk 1 above)
Comment (nathanshelly) [reply totes]: what do you think of refactoring this onto separate lines and changing how we guard the `grep`? something like:

```bash
all_notes="$(ls -At -- "$notes_path"/*.md)"
if [[ -f "$suppressed_files_registry_spec" ]]; then
  # TODO: add context on these flags and/or where this solution came from
  active_notes="$(grep -vFf "$suppressed_files_registry_spec" <<< "$all_notes")"
fi
note_to_render="$(tail -n 1 <<< "${active_notes:-$all_notes}")"
```

File: totes (hunk 2 above)
Comment (glenshelly) [reply totes]: comes from "filespec": https://en.wikipedia.org/wiki/Filespec

Comment (glenshelly) [reply totes]: meant to communicate "this is the full path, and not just the individual file name within the directory"

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+notes/suppressed_files_registry
```
[the remark this answers is not in the document]
Comment (glenshelly) [reply .gitignore]: yes, that does seem better

File: totes (hunk 2 above)
Comment (glenshelly) [reply totes]: and also as used twice in this line you added in an earlier commit: https://github.com/nathanshelly/totes/blob/44880f4e83bc0150423a26441b7d2006ff5cf168/totes#L26

File: totes (hunk 1 above)
Comment (glenshelly) [reply totes]: I do like that better.  Knowing there are other PR's that touch these lines "in the queue", I'd rather wait for the dust to settle slightly, and make this suggested change in a later PR

File: totes [hunk 5]
```diff
@@ -9,6 +9,15 @@ get_render_command() {
   fi
 }
 
+get_notes_path() {
+  echo "${XDG_DATA_HOME:-$HOME/.local/share}/totes"
```
[the remark this answers is not in the document]
Comment (glenshelly) [reply totes]: yes, will do

File: totes [hunk 6]
```diff
@@ -9,6 +9,15 @@ get_render_command() {
...
+}
+
+get_suppressed_files_registry_spec() {
+  notes_dir="$(get_notes_path)"
```
[the remark this answers is not in the document]
Comment (glenshelly) [reply totes]: done (and got it)

File: totes [hunk 7]
```diff
@@ -9,6 +9,15 @@ get_render_command() {
...
+  echo "$notes_dir/suppressed_files_registry"
```
[the remark this answers is not in the document]
Comment (glenshelly) [reply totes]: done

File: .gitignore (hunk 4 above)
[the remark this answers is not in the document]
Comment (glenshelly) [reply .gitignore]: changed

File: totes [hunk 8]
```diff
@@ -21,30 +30,42 @@ get_note_to_render() {
...
     # shellcheck disable=2012
-    note_to_render="$(ls -At -- "$notes_path"/*.md | tail -n 1)"
+    # shellcheck disable=2010
+    note_to_render="$(ls -At -- "$notes_path"/*.md | grep -vFf "$suppressed_files_registry_spec" | tail -n 1)"
     # update last read time to cycle through notes
   fi
   echo "$note_to_render"
 }
 
+remove_file() {
```
[the remark this answers is not in the document]
Comment (glenshelly) [reply totes]: yes, done

File: totes (hunk 3 above)
Comment (glenshelly) [reply totes]: interesting.  Removed.

File: totes (hunk 1 above)
Comment (nathanshelly) [reply totes]: In general I think needing to rebase existing PRs on new changes from other merges should be an expected, not uncommon practice. For example I merged a few PRs today that are creating conflicts and will require a rebase of this PR on `master`. If you'd rather do it in a follow-up though that's totally fine by me.

File: totes [hunk 9]
```diff
@@ -9,32 +9,49 @@ get_render_command() {
...
+    note_to_render="$(ls -At -- "$notes_path"/*.md | grep -vFf "$suppressed_notes_registry_spec" | tail -n 1)"
     # update last read time to cycle through notes
   fi
-  echo "$note_to_render"
+  printf "%s" "$note_to_render"
+}
+
+suppress_note() {
+  # Consider only writing file if it's not already in suppressed_notes
+  echo "$1" >> "$(get_suppressed_notes_registry_spec)"
```
Comment (nathanshelly) [inline totes]: Agree we should check this (could be in a follow-up PR if you prefer for any reason), should be as simple as:

```
grep -sq "$1" "$(get_suppressed_notes_registry_spec)" || {
  echo "$1" >> "$(get_suppressed_notes_registry_spec)"
}
```

**Pull request merged**